flag_test.go

```diff
@@ -356,3 +356,33 @@ func TestNoInterspersed(t *testing.T) {
 		t.Fatal("expected interspersed options/non-options to fail")
 	}
 }
+
+func TestTermination(t *testing.T) {
+	f := NewFlagSet("termination", ContinueOnError)
+	if f.Parsed() {
+		t.Error("f.Parse() = true before Parse")
+	}
+	arg1 := "ls"
+	arg2 := "/home"
```

Comment:
I imagined this might be the case (due to more bugs in handling of `--`) but testing with something that starts with a `-` is the "interesting" case and the presence of this test should make sure regressions won't be introduced. Thank you for enhancing this test!